File: Assets/Resources/howl/LuaModules/Audio/AudioModule.lua
AudioModule = {}
local test = nil


function AudioModule.new()
    return AudioModule
end

function AudioModule:PlayAudio(clipname)
    AudioServiceProxy.PlayAudioClip(clipname)
end

function AudioModule:SetVolume(volume)
    AudioServiceProxy.SetVolume(volume)
end

function AudioModule:StopPlayingAudio()
    AudioServiceProxy.StopPlayingAudio()
end

return AudioModule

File: Assets/LuaBridge/Unity/Scripts/LuaBridgeModules/AudioModule/AudioModule.cs
using System.Threading.Tasks;
using LuaBridge.Unity.Scripts.LuaBridgeServices.AudioService.Interface;

namespace LuaBridge.Unity.Scripts.LuaBridgeModules.AudioModule
{
    public class AudioModule
    {
        private readonly IAudioService audioService;

        public AudioModule(IAudioService audioService)
        {
            this.audioService = audioService;
        }

        public async Task PlayAudio(string audioClip)
        {
            await audioService.PlayAudioClip(audioClip);
        }
        

        public void SetVolume(int volume)
        {
            audioService.SetVolume(volume);
        }

        public void StopPlayingAudio()
        {
            audioService.StopPlayingAudio(); 
        }
    }
}